### ISSUE TITLE
http_proxy.rb -- invalid request URI

### ISSUE THREAD

**Issue by do11** [igrigorik/em-proxy#51]
Unfortunately, `http_proxy.rb` example have bug when used with HTTP/1.0 servers, because, it does not strip `http://server` prefix from the proxied HTTP request. As a consequence, some web servers return `400 Bad Request` `invalid request URI error`.

When HTTP proxy requested with `GET http://server/uri HTTP/1.0` it *should* pass to real _server_ `GET /uri HTTP/1.0`. [RFC1945](https://tools.ietf.org/html/rfc1945): _"The absoluteURI form is only allowed when the request is being made to a proxy."_ Thus, to real servers _absoluteURI_ is not allowed.

**Comment by igrigorik** [comment]
FWIW, http_proxy.rb is intended as a demo, not a production-ready proxy server. That said..

Seems like the issue here is that your client is initiating an HTTP/1.0 request? The proxy doesn't know what protocols the server supports, it just relays the data from the client.

**Comment by do11** [comment]
It does not work even for HTTP/1.1 (_foobar2000_ player), but server is _Icecast Streaming Media Server_, which is popular across internet radios (http://dir.xiph.org/). That's only reason I tried em-proxy is that Webrick does not support streaming at all.

Only RFC's prohibition I found (after not long search) is for HTTP/1.0. Maybe absoluteURI _is_ permitted for HTTP/1.1, maybe, and it's just Icecast bug. Possible. But, squid, AFAIK, is always stripping absolute part from URL even for HTTP/1.1.

> FWIW, http_proxy.rb is intended as a demo, not a production-ready proxy server. 

So there is lower standard for learning material? OK

:running:
